lib/ethpy/ethpy/hyperdrive/api.py

```diff
@@ -760,107 +847,58 @@ def get_max_long(self, budget: FixedPoint) -> FixedPoint:
-            governance=self._contract_pool_config["governance"],
-            feeCollector=self._contract_pool_config["feeCollector"],
-            Fees=Fees(
-                curve=str(self._contract_pool_config["fees"][0]),
-                flat=str(self._contract_pool_config["fees"][1]),
-                governance=str(self._contract_pool_config["fees"][2]),
-            ),
-            oracleSize=str(self._contract_pool_config["oracleSize"]),
-            updateGap=str(self._contract_pool_config["updateGap"]),
-        )
-        pool_info_str = PoolInfo(
-            shareReserves=str(self._contract_pool_info["shareReserves"]),
-            shareAdjustment=str(self._contract_pool_info["shareAdjustment"]),
-            bondReserves=str(self._contract_pool_info["bondReserves"]),
-            lpTotalSupply=str(self._contract_pool_info["lpTotalSupply"]),
-            sharePrice=str(self._contract_pool_info["sharePrice"]),
-            longsOutstanding=str(self._contract_pool_info["longsOutstanding"]),
-            longAverageMaturityTime=str(self._contract_pool_info["longAverageMaturityTime"]),
-            shortsOutstanding=str(self._contract_pool_info["shortsOutstanding"]),
-            shortAverageMaturityTime=str(self._contract_pool_info["shortAverageMaturityTime"]),
-            withdrawalSharesReadyToWithdraw=str(self._contract_pool_info["withdrawalSharesReadyToWithdraw"]),
-            withdrawalSharesProceeds=str(self._contract_pool_info["withdrawalSharesProceeds"]),
-            lpSharePrice=str(self._contract_pool_info["lpSharePrice"]),
-            longExposure=str(self._contract_pool_info["longExposure"]),
-        )
+        pool_config_str = self._serialized_pool_config()
+        pool_info_str = self._serialized_pool_info()
         # pylint: disable=no-member
-        max_short = pyperdrive.get_max_short(
-            pool_config_str,
-            pool_info_str,
-            str(budget.scaled_value),
-            pool_info_str.sharePrice,
-            checkpoint_exposure=str(self.latest_checkpoint["longExposure"].scaled_value),
-            maybe_conservative_price=None,
-            maybe_max_iterations=None,
+        max_short = FixedPoint(
+            scaled_value=int(
+                pyperdrive.get_max_short(
+                    pool_config_str,
+                    pool_info_str,
+                    str(budget.scaled_value),
+                    pool_info_str.sharePrice,
+                    checkpoint_exposure=str(self.latest_checkpoint["longExposure"].scaled_value),
+                    maybe_conservative_price=None,
+                    maybe_max_iterations=None,
+                )
+            )
         )
-        return FixedPoint(scaled_value=int(max_short))
+        bonds_in = self.get_in_for_out(amount_out=max_short / self.pool_config["sharePrice"], shares_out=True)
+        curve_fee, _, _ = self.calculate_fees_out_given_bonds_in(bonds_in=bonds_in)
```

Comment:
We shouldn't need to do this if you're using the `get_max_short` from the Rust SDK. That function already takes fees into account. When I mentioned that we should "make sure that we're taking fees into account" I was only talking about the function that calculates the bond amount required to bring the rate to a targe.